indico_toolkit/indico_wrapper/indico_wrapper.py

```diff
@@ -1,18 +1,50 @@
-from typing import List, Dict, Union
+from typing import List, Union
 from indico.queries import (
     RetrieveStorageObject,
     GraphQLRequest,
     JobStatus,
     CreateModelGroup,
     ModelGroupPredict,
+    CreateStorageURLs
 )
 from indico.types import Dataset, ModelGroup
 from indico import IndicoClient
+from indico.errors import IndicoRequestError
+from functools import wraps
+import time
 
 from indico_toolkit.types import Predictions
 from indico_toolkit import ToolkitStatusError
 
 
+def retry(exceptions, wait=0.5):
+    """
+    Decorator for retrying functions after specified exceptions are raised
+    Args:
+    exceptions (Exception or Tuple[Exception]): exceptions that should be retried on
+    wait (float): time in seconds to wait before retrying
+    num_retries (int): the number of times to retry, this must be passed into the decorated function as a kwarg 
+    """
```

Comment:
I think we should just use a default # of retries and take that away from the user (or make it a kwarg on 'retry' so it can be specified specific to what it wraps). It seems like a bunch of the methods you are wrapping don't have that kwarg anyways